src/components/redux/books/booksReducer.js

```diff
@@ -1,25 +1,67 @@
-import bookState from './books-state';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
 
-const ADD_BOOK = 'ADD_BOOK';
-const REMOVE_BOOK = 'REMOVE_BOOK';
+const getURL = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/GorvPCNNZt9xnS7jfgYY/books/';
+const request = new Request(getURL);
 
-/** reducer */
-export default function bookReducer(state = bookState, action) {
-  switch (action.type) {
-    case ADD_BOOK:
-      return [
-        ...state,
-        {
-          key: action.payload.key,
-          categorie: 'Islamic',
-          title: action.payload.title,
-          author: action.payload.author,
-        },
-      ];
+export const getBooks = createAsyncThunk(
+  'book/getBooks',
+  async () => fetch(getURL).then((res) => res.json()),
+);
```

Comment:
- [ ] Checking once more on the requirements, the action creator for fetching books should:
- should `dispatch` a new action type that will pass the books list to the reducer and update the state. This need to be done here when fetching the book. Kindly look into this. 
- This should be the same with the other action creators in this file. That's `sendBook`, `removeBook`

Reply:
Do you mean using a middleware?

Reply:
> Do you mean using a middleware?

Hi @maenkhraisha, you are already making use of middleware. I'm aware you are dispatching the action type in another file. But the requirements asked us to dispatch the action type right away when creating the action creator. I just need to see the dispatch keyword when creating your action creators.

Reply:
hi @FrederickMih you mean like this 
  dispatch(sendBook({
      itemId, category, title, author,
    }), [**dispatch**]);